raster-foundry/raster-foundry
# Upgrade to GT 2.0

## Overview

This upgrades Raster Foundry to use the latest release of GeoTrellis (2.0). Also upgrades MAML to a later release as well (still depends on an RC for 2.0).

A fix for a change in behavior around listing zoom level 0 attributes and layers is included in the second commit here -- which prevented the upgrade previously. That is primarily what the testing instructions revolve around verifying, but it would be good to test the rest of the application as well.

### Checklist

## Testing Instructions

 * Recompile/assemble the batch subproject:
```
./scripts/console api-server ./sbt
project batch
assembly
```
 * Delete attributes for a layer in the Raster Foundry database:
```
./scripts/psql
delete from layer_attributes where layer_name = 'a9453bed-67ae-47c3-8320-25b6112c95df'
```
 * Rerun `s3` attribute migration for layer
```
./scripts/console batch "java -cp /opt/raster-foundry/jars/rf-batch.jar com.azavea.rf.batch.Main migration_s3_postgres s3://rasterfoundry-staging-catalogs-us-east-1/layers layer_attributes a9453bed-67ae-47c3-8320-25b6112c95df"
```
 * Verify the number of entries in the layer attributes table is equal to the number deleted previously (should be 73):
```
./scripts/psql
select count(*) from layer_attributes where layer_name = 'a9453bed-67ae-47c3-8320-25b6112c95df'
```
 * To test that Sentinel imports work once again:
```./scripts/console batch "java -cp /opt/raster-foundry/jars/rf-batch.jar com.azavea.rf.batch.Main import_sentinel2 2018-08-13"```

Closes #3826
Closes #3828
